index.js

```diff
@@ -112,19 +112,39 @@ export const createQueryParameters = ({
 };
 
 export default function open(params) {
-	createOpenLink(params)();
+	createOpenLink(params)()
 }
 
 export function createOpenLink({ provider, ...params }) {
 	// Returns a delayed async function that opens when executed
-	if (!provider) {
-		defaultProvider = (Platform.OS === 'ios') ? 'apple' : 'google';
+	return async () => {
+		
+		let mapProvider = provider;
+		if (!provider) {
+			// use default provider
+			mapProvider = (Platform.OS === 'ios') ? 'apple' : 'google';
+		} else {
+
+			if (Platform.OS === 'ios' && provider === 'google') {
+				// Device is iOS with google maps as provider
+				try {
+					// check that device can open google maps. Else default to apple maps
+					const canOpen = await Linking.canOpenURL('comgooglemaps://?center=40.765819,-73.975866');
+					if (canOpen) { 
+						mapProvider = 'google'
+					} else { 
+						throw new Error('Cannot open google maps, falling back to apple maps');
+					}
+				} catch (error) {
+					mapProvider = 'apple'
+					console.warn('Open google maps error: ', error)
+				}
+			}
+		}
+		// Allow override provider, otherwise use the default provider
+		const mapLink = createMapLink({ provider: mapProvider, ...params });
+		return await Linking.openURL(mapLink).catch(err => console.error('An error occurred', err));
```

Comment:
I had to put all the method body in the delayed function, in order to achieve the same functionality as before for `createOpenLink`. I have tested this and it works on both platforms